packages/@react-spectrum/dialog/src/Dialog.tsx

```diff
@@ -81,7 +81,7 @@ function Dialog(props: SpectrumDialogProps, ref: DOMRef) {
   }
 
   return (
-    <FocusScope contain restoreFocus>
+    <FocusScope contain restoreFocus autoFocus>
```

Comment:
I was worried that this might break buttons in the dialog with autoFocus but doesn't seem to be the case from testing

Reply:
`useAutoFocus` within `FocusScope` is designed to first test to see if something within the scope already has focus, and only set focus if that test returns false.

https://github.com/adobe/react-spectrum/blob/1bb3bf2e625bb3d259fc0d17126e692efbef44f8/packages/%40react-aria/focus/src/FocusScope.tsx#L378-L389